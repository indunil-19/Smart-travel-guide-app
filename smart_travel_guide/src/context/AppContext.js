import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}