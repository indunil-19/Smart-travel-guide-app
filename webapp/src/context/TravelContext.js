import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 