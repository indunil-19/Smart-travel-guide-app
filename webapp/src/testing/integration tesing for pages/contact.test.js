import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import { createMemoryHistory } from 'history'
import Contact from "../../screens/TravelPlanApp/contact"

describe('Contact us page', () => {
    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><Contact/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><Contact  /></TravelProvider></MemoryRouter>,div)
    })

    

})
