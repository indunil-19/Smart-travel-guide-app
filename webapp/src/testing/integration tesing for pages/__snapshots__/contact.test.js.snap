// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact us page should match snapshot 1`] = `
<div
  className="css-1y0wkpi"
>
  <ins
    className="chakra-heading css-1knigk4"
  >
    Contact Us
  </ins>
  <p
    className="chakra-text css-1n49pl7"
  >
    You can contact us through our facebook page or whatsapp
  </p>
  <div
    className="chakra-stack css-84zodg"
  >
    <iframe
      allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
      allowfullscreen="true"
      frameborder="0"
      height="500"
      scrolling="no"
      src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId"
      style={
        Object {
          "border": "none",
          "overflow": "hidden",
        }
      }
      width="340"
    />
  </div>
</div>
`;
